### ISSUE TITLE
[project] Learned animal-vs-FP classifier — replace mean-brightness treadmill

### ISSUE THREAD

**Issue by odanree** [odanree/wildlife-detector#137]
## Goal

Replace `NIGHT_INSECT_BRIGHTNESS_MIN` threshold treadmill with a learned binary classifier that outputs `P(animal | bbox_features)`. Scene-invariant — handles rat-behind-branch, cat-on-bright-bricks, moth-on-dark-ground without hand-tuned thresholds.

## Why

Current filter has been ratcheted 100 → 110 → 115 → 130 as more edge cases surface. Each threshold bump costs 5-15% of FP catch rate. Root cause: MOG bbox mean is confounded by any bright pixels the bbox happens to include (surrounding bricks, foreground branches). Signal itself is not clean; no single threshold works across scenes.

## Data (ready today)

~4200 labeled examples in Postgres alerts table:
- 1022 positive (real animal)
- 3165 negative (FP)
- Per-camera / per-species breakdown captured
- Rodent 911, cat 80, squirrel 19, raccoon 9, bird+dog 1

## Design

**Model**: LightGBM/CatBoost gradient boosting on tabular features. Not deep learning — data volume + interpretability + inference cost all favor GBT.

**Features** (per bbox at DECISION time):
- Tabular: tight_mean/max/min/ar/area, wide_mean/max/min/ar, baseline_diff_score, mog_track_age, merge_cluster_size, camera_id, hour_of_day, brightness_mode
- Optional CNN embedding (MobileNetV3-Small pretrained, 16-dim) on wide-crop for visual context

**Split**: by track_id + timestamp bucket (not by row) to prevent leakage. 70/15/15 train/val/test.

**Metrics**:
- Primary: recall on real_rodent / real_cat / real_raccoon (must not clip real animals)
- Secondary: FP rejection rate vs current threshold=130
- Per-camera recall targets ≥ 95%
- Named failure classes evaluated separately: rat-behind-branch, cat-on-bricks, moth-on-dark

## Integration

New env `INSECT_FILTER_MODE=classifier|threshold` (default `threshold` for rollback safety).

```python
if INSECT_FILTER_MODE == "classifier":
    features = _extract_features(stats, det, baseline, ...)
    p_animal = _clf.predict_proba([features])[0][1]
    if p_animal < FILTER_PROB_THRESHOLD:  # env, default 0.2
        continue  # classified as FP, skip VLM
```

Model file baked into detector Docker image (~5 MB). Retrain script `scripts/train_animal_vs_fp.py` reads DB → exports snapshots → trains → saves `models/`.

## Rollout

1. **Ship code + model, MODE=threshold** — no behavior change, infrastructure only
2. **Shadow mode**: log classifier prediction alongside current filter decision. Compare on a full night. Zero risk.
3. **Per-camera enable**: flip backyard first (highest FP volume). Watch overnight. Keep if recall ≥ threshold + FP catch better.
4. **Repeat yard, rooftop.**

Instant rollback: env flip.

## Tasks

- [ ] `scripts/train_animal_vs_fp.py` — DB export + snapshot fetch + baseline crop + feature extraction + train + save model (4h)
- [ ] `src/detection/animal_classifier.py` — load model + `predict(features) -> float` API (2h)
- [ ] `src/pipeline.py` — env dispatch `INSECT_FILTER_MODE`, feature-extract call in insect gate (2h)
- [ ] Shadow-mode logging: log both filter decisions per DECISION line for comparison (1h)
- [ ] `scripts/compare_filter_modes.py` — parse shadow logs, compute confusion matrix vs labels (2h)
- [ ] Docker image update — add lightgbm/catboost + model file to detector image (1h)
- [ ] Validation night — shadow-mode on all 3 cameras, review confusion matrix (overnight + 2h analysis)
- [ ] Enable classifier per-camera with monitoring
- [ ] Document env vars in .env.example + README

**Total: ~2 focused days of work + 1-2 nights of validation.**

## Alternatives not chosen

- Deep CNN from scratch: cat/squirrel/raccoon volume too low for multi-class
- Full YOLO fine-tune on IR night imagery: 10-100× more bbox labels needed
- More hand-tuned rules stacked: what we've been doing; classifier subsumes with learned weights

## Risks

- Systematic label bias (mostly-easy labels correct, hard ones marked unclear) → classifier learns same bias. Mitigation: audit 50 random borderline labels before training.
- Snapshot files missing for old alerts → some rows unusable. Mitigation: filter to alerts with snapshot_path.exists().
- Docker image bloat if CNN embedding added → skip embedding, tabular-only fallback. Still workable.

## Success criteria for shipping to production

- Test-set recall ≥ 98% on `real_rodent`, ≥ 95% on `real_cat`
- FP rejection rate at least 5pt better than threshold=130 filter
- Per-camera recall parity (no camera regresses)
- Shadow mode validates offline metrics on a full night of live data

## Related

- Deferred from PR #133 discussion (raised NIGHT_INSECT_BRIGHTNESS_MIN 115→130)
- Would resolve the "rat behind branch" class of failure seen on alert 34949
- Would eliminate the "spider web animated FP" class our stationary suppression can't handle
